src/cookbooks/main_ping_exponential_histograms.md

```diff
@@ -0,0 +1,312 @@
+
+We can reproduce this approach by using the [`histogram_normalize`](https://github.com/mozilla/bigquery-etl/blob/master/udf/histogram_normalize.sql) UDF, which explicitly takes a set of histograms and makes sure that the values for each one don't sum up to any more than one:
+
+```sql
+DECLARE four_twenty DEFAULT DATE('2020-04-20');
+
+WITH per_build_client_day AS
+  (SELECT PARSE_DATETIME("%Y%m%d%H%M%S", application.build_id) AS build_id,
+          client_id,
+          `moz-fx-data-shared-prod`.udf.histogram_normalize(
+            `moz-fx-data-shared-prod`.udf.histogram_merge(
+                ARRAY_AGG(`moz-fx-data-shared-prod`.udf.json_extract_histogram(
+                    payload.histograms.FX_TAB_SWITCH_SPINNER_VISIBLE_MS)
+                )
+            )
+          ) AS tab_switch_visible_ms
+   FROM `moz-fx-data-shared-prod`.telemetry.main
+   WHERE application.channel='nightly'
+     AND normalized_os='Windows'
+     AND application.build_id > FORMAT_DATE("%Y%m%d", DATE_SUB(four_twenty, INTERVAL 14 DAY))
+     AND application.build_id <= FORMAT_DATE("%Y%m%d", four_twenty)
+     AND DATE(submission_timestamp) >= DATE_SUB(four_twenty, INTERVAL 14 DAY)
+     AND DATE(submission_timestamp) <= four_twenty
+   GROUP BY build_id,
+            client_id),
+     merged_histograms AS
+  (SELECT build_id,
+          KEY,
+          SUM(value) AS value,
+   FROM per_build_client_day,
+        UNNEST(per_build_client_day.tab_switch_visible_ms.
+               VALUES)
+   GROUP BY KEY,
+            build_id),
+     as_struct AS
+  (SELECT build_id,
+          STRUCT(ARRAY_AGG(STRUCT(KEY, value)) AS
+                 VALUES) AS spinner_visible_long_ms
+   FROM merged_histograms
+   GROUP BY build_id),
+     percentiles AS
+  (SELECT build_id,
+          `moz-fx-data-shared-prod`.udf.histogram_percentiles(spinner_visible_long_ms, [ .05,
+      .25,
+      .5,
+      .75,
+      .95]) AS percentile_nested
+   FROM as_struct)
+
+SELECT build_id,
+       percentile,
+       value
+FROM percentiles
+CROSS JOIN UNNEST(percentiles.percentile_nested);
+```
+
+[link](https://sql.telemetry.mozilla.org/queries/71489/source)
+
+You'll notice that, in addition to the call to `histogram_normalize`, this query groups by `client_id` in addition to `build_id`: this is necessary, otherwise we would be normalizing by groups of clients sharing the same `build_id` which doesn't really make any sense.
+
+In any case, the result of this query is this graph:
+
+![](https://i.imgur.com/cX3deIM.png)
+
+Things are looking much better! The 95th percentile is still capped out at 1000, but the other percentiles are much lower.
+
+## Slicing along arbitrary dimensions
```

Comment:
A "compare and contrast" across two different dimensions in a single time period makes more sense than a "change over time". It would be easier to interpret a bar graph where two populations can be compared directly by the height of their percentiles.

Reply:
Hmm, I see your point but am not sure if I agree. If you were to look at a bulk aggregated view, it wouldn't be clear that the difference was sustained. Am inclined to leave this as-is -- hopefully it won't be hard for people to modify these queries/examples into whatever visualization they prefer.